README.md

```diff
@@ -41,5 +41,9 @@ Otherwise, your script for executing specs may look something like:
 chmod +x test-splitter
 ./test-splitter # fetches the test plan for this node, and then executes the rspec tests
 ```
-
-
+### Exit code
+| Exit code | Description |
+| ---- | ---- |
+| 0 | All tests pass |
+| 1 | At least one test fails |
```

Comment:
Do `0`/`1` specifically mean exactly these things? Or do they mean whatever the underlying test runner means?

Could it be expressed like this?

- `0`: success (passed through from test runner)
- `1`: failure (passed through from test runner)
- `3`: Test Splitter failure (eg. config error)
- `*`: other errors (passed through from the test runner)

(I guess for now we have a good understanding of RSpec and its exit statuses, but that might be quite as true for future supported test runners)

Given that the above list is basically “exit codes are always passed through from the test runner, but there's at least one special one that we generate for internal problems”, I wonder if we should reserve a range of not-commonly-used exit statuses for our internal use, e.g. 16–23.

Reply:
This comment sounds reasonable, yep. The 16–23 exit code reservation would include changing the current `3` code to `16` is that right?

Reply:
Yes, we run the rspec tests in a subprocess and exit the main program the same code returned by the subprocess. 0, 1 and other errors are passed through from the test runner. The only special case is the test splitter internal errors (currently 3).

I'll create a PR to improve the README, and update the exit code for internal failure to 16. 

I'm also curious why we pick 16-23? I'm assuming exit codes below 16 could be used commonly for other exit reasons?

Reply:
@pda any reasoning behind the 16–23?